### ISSUE TITLE
strange shape question

### ISSUE THREAD

**Issue by flynnamy** [TimoBolkart/TF_FLAME#45]
Hi，I use accurate 2D landmark to fit and add the weight to let it close eyes.It shows that it can close eyes but get strange shape. I try to add regularizer of shape and expression, but it does not work.Any suggestions? @TimoBolkart 
![image](https://user-images.githubusercontent.com/20398890/103971909-bcddb700-51a6-11eb-9ded-8bab1b9e91b8.png)
![image](https://user-images.githubusercontent.com/20398890/103971918-c535f200-51a6-11eb-84d1-08f888f3ec84.png)

**Comment by TimoBolkart** [comment]
Can you share the set of landmarks? Then I can have a look into it.

**Comment by flynnamy** [comment]
Hi,please try this landmarks file! @TimoBolkart 
[wink.zip](https://github.com/TimoBolkart/TF_FLAME/files/5829868/wink.zip)

**Comment by flynnamy** [comment]
Hi, do you try this landmark file?Any suggestions? @TimoBolkart

**Comment by TimoBolkart** [comment]
Sorry for the late response. I did play with the landmarks a bit. The reason for the strage shape is that the model priors of shape / expression are too low, so the landmarks are tightly fit, but the reconstructed shape is unrealistic. 

One thing to get better eye blinks reconstruction, you can use an additional loss to measure the distance between the upper / lower eyelid landmarks for the projected model landmarks and the target landmarks, and minimizing the difference between them. The loss (i.e. `lmk_eye_dist`) would be something like that:

'model_lmks_dist = tf.subtract(tf.gather(lmks_proj_2d, [20, 21, 26, 27]), tf.gather(lmks_proj_2d, [24, 23, 30, 29]))`
`target_lmks_dist = target_2d_lmks[[20, 21, 26, 27]] - target_2d_lmks[[24, 23, 30, 29]]`
`lmk_eye_dist = 1.0*tf.reduce_sum(tf.square(tf.subtract(model_lmks_dist, target_lmks_dist))) / (factor ** 2)`

Another problem is that while FLAME is able to blink with the eyelids, but the blink of both eyes is quite correlated in the training data. Therefore, blinking only with one eye is difficult. You could use these additive blendshapes for [left eye](http://files.is.tue.mpg.de/tbolkart/FLAME/l_eyelid.npy) and [right eye](http://files.is.tue.mpg.de/tbolkart/FLAME/r_eyelid.npy) with weight [0,1] to close either eye for the FLAME template.
